node/node.go

```diff
@@ -870,6 +901,21 @@ func (n *Node) GetAccount(address string, options api.AccountQueryOptions) (api.
 	}, blockInfo, nil
 }
 
+func adaptBlockInfoIfError(err error) (api.BlockInfo, error) {
+	if err == nil {
+		return api.BlockInfo{}, nil
+	}
+
+	apiBlockInfo, ok := extractApiBlockInfoIfErrAccountNotFoundAtBlock(err)
+	if ok {
+		return apiBlockInfo, nil
+	}
+	if err == ErrCannotCastAccountHandlerToUserAccountHandler {
```

Comment:
Is this ok? What happens if you provide an encoded code-hash instead of an address?
The `ErrCannotCastAccountHandlerToUserAccountHandler` is returned both if the account handler is nil or it is not of type userAccountHandler. Don't we want to signal an error in the case the loaded account is of bad type?

Reply:
If we provide a non-address an error will be returned because of the bech32 validation.
@andreibancioiu do you think about other situations where this error might be thrown and it really indicates an issue?